### PR TITLE
Deprecate the ugly perspective -JG syntax and introduce a modern form

### PR DESCRIPTION
**Pull request by PaulWessel**
While this is backwards compatible, the new form uses optional modifiers.  This PR updates the documentation, the example script in the cookbook, but have not changed the test scripts (yet) to make sure both forms work.

**New syntax**

**-Jg**|**G**_lon0/lat0_/_scale_|_width_[**+a**_azimuth_][**+t**_tilt_][**+v**_vwidth_/_vheight_][**+w**_twist_][**+z**_altitude_]

**Old syntax**

**-Jg**|**G**_lon0/lat0/altitude/azimuth/tilt/twist/Width/Height/scale|width_

I have added three new tests that explores variations in azimuth, altitude, and view window.  I will add similar tests for twist and tilt later.
The documentation, doc examples, doc scripts and animations using the perspective projection have been updated to reflect the new syntax.  We will leave old tests scripts for backwards testing and add new scripts using the modernized format.

### CODE REVIEW

**Review by Esteban82 — APPROVED** [review]

**Comment by PaulWessel** [comment]
While @Esteban82 has approved this one (since it works), if anyone has comments or changes to the particular choices of modifiers then now is a good time.  FYI, here is the three test results for varying altitude, azimuth, and view restriction:

**Altitude:**
altitude

**Azimuth:**
azimuth

**View:**
view

File: doc/rst/source/explain_-J.rst_ [hunk 1]
```diff
@@ -205,15 +205,19 @@ respectively):
...
-        *lon0/lat0* specifies the projection center. *altitude* is the height (in km) of the viewpoint above local sea
-        level. If *altitude* is less than 10, then it is the distance from the center of the earth to the viewpoint in
-        earth radii. If *altitude* has a suffix **r** then it is the radius from the center of the earth in kilometers.
-        *azimuth* is measured to the east of north of view. *tilt* is the upward tilt of the plane of projection. If
-        *tilt* is negative, then the viewpoint is centered on the horizon. Further, specify the clockwise *twist*,
-        *Width*, and *Height* of the viewpoint in degrees. Give *scale* (with **-Jg**; as **1:**\ *xxxx* or
-        *radius/lat*, where *radius* is distance in :ref:`plot-units <plt-units>` from origin to the oblique latitude
-        *lat*) or width (with **-JG**; in :ref:`plot-units <plt-units>`).
+    **-Jg**\|\ **G**\ *lon0/lat0*\ [*/horizon*]\ */*\ *scale*\|\ *width*\ [**+a**\ *azimuth*][**+t**\ *tilt*][**+v**\ *vwidth/vheight*][**+w**\ *twist*][**+z**\ *altitude*] (General Perspective).
```
Comment (meghanrjones) [inline doc/rst/source/explain_-J.rst_:208]: What is the optional [*/horizon] here? I do not see it in the usage message of description.

Comment (PaulWessel) [reply doc/rst/source/explain_-J.rst_]: Yep, should not be there.  Now gone.

File: doc/rst/source/cookbook/map-projections.rst [hunk 2]
```diff
@@ -404,24 +404,27 @@ Perspective projection (**-Jg** **-JG**)
 
 **Syntax**
 
-    **-Jg**\|\ **G**\ *lon0/lat0/altitude/azimuth/tilt/twist/Width/Height/*\ *scale*\|\ *width*
+    **-Jg**\|\ **G**\ *lon0/lat0*\ [*/horizon*]\ */*\ *scale*\|\ *width*\ [**+a**\ *azimuth*][**+t**\ *tilt*][**+v**\ *vwidth/vheight*][**+w**\ *twist*][**+z**\ *altitude*]
```
Comment (meghanrjones) [inline doc/rst/source/cookbook/map-projections.rst:407]:
```suggestion
    **-Jg**\|\ **G**\ *lon0/lat0*\ */*\ *scale*\|\ *width*\ [**+a**\ *azimuth*][**+t**\ *tilt*][**+v**\ *vwidth/vheight*][**+w**\ *twist*][**+z**\ *altitude*]
```

**Review by meghanrjones — COMMENTED** [review]
New syntax looks nice. I did not test it since that's already been done.

**Comment by Esteban82** [comment]
Is this supposed to work this way? I get a flip image when I add +du to the width

`gmt grdimage -JG-73/40/20c+du+z160+a210+t55+v36 -Rg  @earth_relief_02m -png Test `


Test

**Pull request merged**